pyangbind/lib/yangtypes.py

```diff
@@ -169,7 +169,11 @@ def __init__(self, *args, **kwargs):
         self.__check(args[0])
       except IndexError:
         pass
-      super(RestrictedClass, self).__init__(*args, **kwargs)
+
+      try:
+        super(RestrictedClass, self).__init__(*args, **kwargs)
+      except TypeError:
```

Comment:
Can you comment on why this is required? I think that there was a previous discussion that in some cases in Python 3 `__init__ ` doesn't take an argument. What does this mean if there is a value specified for this class as it is initialised - since I think in most cases this was how we were handing through the value.

Reply:
Honestly, this piece of code is kind of obscure and I don't recall corretly. If I recall correctly some of the classes generated by pyangbind are inherited by what python considers a `type` not a `class` so the `TypeError` was raised.